### PR TITLE
Add unit tests for AdminMenu component

### PR DESCRIPTION
**Pull request by DhiraPT**
**Summary**
Added comprehensive unit tests for `AdminMenu` component with 100% code coverage. No critical bugs were found in the component.

**Code Review Results**
- No critical bugs found - Component is functionally correct
- No security issues - Simple presentational component with proper React Router usage
- No logic errors - All navigation links work as expected

**Changes Made**
- 30 unit tests with 100% code coverage using React Testing Library
- Tests cover: component rendering, navigation links, active states, accessibility, edge cases, and component isolation
- All tests are pure unit tests (no integration or e2e logic)

### CODE REVIEW

File: client/src/components/AdminMenu.test.tsx [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+      expect(link).toHaveAttribute("href", "/dashboard/admin/orders");
+      expect(link?.classList.contains("list-group-item")).toBe(true);
+      expect(link?.classList.contains("list-group-item-action")).toBe(true);
+    });
+
+    it("renders exactly 4 navigation links", () => {
+      renderWithRouter();
+      const links = screen
+        .getAllByRole("link")
+        .filter((link) => link.classList.contains("list-group-item"));
```
[the remark this answers is not in the document]
Comment (RichDom2185) [reply client/src/components/AdminMenu.test.tsx:96]: Out of scope, issue created.

**Review by RichDom2185 — APPROVED** [review]
LGTM, thanks!
